src/parser/validator.ts

```diff
@@ -0,0 +1,52 @@
+import { RootParser } from "../parser/RootParser";
+import { makeCstVisitor, VisitorResult, CommandNode } from "../parser/makeCstVisitor";
+
+interface validateCSTResult {
+    errors: ValidateCSTError[];
+    hasErrors: boolean;
+}
+
+interface ValidateCSTError {
+    error: Error,
```

Comment:
Why not reuse the VisitorError? I'm thinking we should make the "VisitorError" into the actual "BluehawkError" class. All diagnostics can conform to the same interface from the visitor on.